node/src/node_configurator/configurator.rs

```diff
@@ -554,6 +554,11 @@ impl Configurator {
             "earningWalletAddressOpt",
         )?;
         let start_block = Self::value_required(persistent_config.start_block(), "startBlock")?;
+        let max_block_count_opt =
+            match Self::value_required(persistent_config.max_block_count(), "maxBlockCount") {
+                Ok(value) => Some(value),
+                _ => None,
```

Comment:
Really? An error retrieving from the database (probably meaning the database is corrupt) is converted into an innocent `None` value? Are we sure we want to do this? Almost everywhere else in the code, a corrupt database triggers an instant panic, so that we don't keep running with a bad database and perhaps damage it further.